### PR TITLE
[17.0][MIG] sentry

### PR DESCRIPTION
**Pull request by aisopuro**
Migration itself is not particularly notable. However there are some oddities in this PR:
* Based on the 16.0 branch as usual, up until commit af35006d6c32903d387ff86249e91508e29f7132
* Then cherry-picked a fix from 15 that apparently had not made it to 16. Commits fbfa2e5a39626a742550a200c0979d7161ef3c61 and 9213ab137f6438267331836bdb36635bf172f08f
* Pre-commit was not able to automatically fix everything, so made some manual fixes in a separate commit

### CODE REVIEW

**Comment by aisopuro** [comment]
@moylop260   , @fernandahf   and/or @naglis I understand you're the maintainers of the `sentry` module. My PR seems to be alright except for some code coverage checks failing. Most of the failures seem to be from various helper functions that seem fairly complicated to test.

I wanted to ask if you had any idea how I should improve the coverage: would it be OK for me to either tell coverage to ignore certain functions altogether, or can I improve the coverage of a different module in this same PR to make the tests pass?

Thanks in advance. I hope to be able to work on this if I can just get some guidance, I don't want this PR to grow stale.

**Review by thomaspaulb — APPROVED** [review]
I don't think coverage changes are blocking on migration PR's.

**Comment by aisopuro** [comment]
Rebased to resolve conflicts in `requirements.txt` (another PR was merged that also auto-generated it).

File: sentry/processor.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+            data["extra"] = self.filter_extra(data["extra"])
+
+        if "level" in data:
+            data["level"] = self.filter_level(data["level"])
+
+        return data
+
+    @property
+    def sanitize_keys(self):
+        pass
```
Comment (thomaspaulb) [inline sentry/processor.py:41]: If you want an easy way to increase coverage, you can just edit `test_processor.py` and just add calls to the functions here that are marked as not covered, and run an `assert` on the result. You'd be testing trivial stuff, like, what if the input is None, what if the input is a `bytes` object and not a string, etc. It's like ordering `-1` beers, `1000000` beers etc. But it increases coverage.

**Comment by aisopuro** [comment]
OK, looks like I had forgotten to import the test files :sweat: . All checks are green now.

**Review by atchuthan — APPROVED** [review]
:+1:

File: sentry/tests/test_client.py [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+    """Removes handlers of handlers ignored list."""
+    _IGNORED_LOGGERS.discard(handler_name)
+
+
+class TestException(exceptions.UserError):
+    pass
+
+
+class InMemoryTransport(HttpTransport):
+    """A :class:`sentry_sdk.Hub.transport` subclass which simply stores events in a list.
```
Comment (thomaspaulb) [inline sentry/tests/test_client.py]: Excuse the bureaucracy here, but apparently this has to be truncated from 89 to 88 lines according to new [ruff](https://github.com/OCA/server-tools/actions/runs/8970280465/job/24633463815) rules.

Comment (aisopuro) [reply sentry/tests/test_client.py]: Rebased and fixed.

**Pull request merged**